App.js

```diff
@@ -48,18 +93,94 @@ export default function App() {
 
   // press Reset button
   const pressReset = useCallback(() => {
-    setGameState(() => {
+    setGameState(prevGameState => {
+      let workNumberOfPlayers = prevGameState.numberOfPlayers;
+      let workMessage = 'Roll again';
+      let workOptMessage = 'Player 1';
+      let workAbout = prevGameState.about;
+      let workTurnTotals = [1,0];
+      let workWhoseTurn = 1;
+      if (workNumberOfPlayers === 1 && prevGameState.position1 === 0) {
+        workNumberOfPlayers = 0;
+        workMessage = '1-player Game';
+        workOptMessage = '2-player Game';
+        workTurnTotals = [0,0];
+        workWhoseTurn = 0;
+        workAbout = true;
+      }
+      if (workNumberOfPlayers === 2 && prevGameState.position1 === 0 && prevGameState.position2 === 0) {
+        workNumberOfPlayers = 0;
+        workMessage = '1-player Game';
+        workOptMessage = '2-player Game';
+        workTurnTotals = [0,0];
+        workWhoseTurn = 0;
+        workAbout = true;
+      }
       return {
         roll: 0,
         even: false,
         odd: false,
         position1: 0,
-        message: 'Roll again',
-        optMessage: 'Player',
+        position2: 0,
+        message: workMessage,
+        optMessage: workOptMessage,
         score: 0,
         board: JSON.parse(JSON.stringify(newBoard)),
         endOfGame: false,
+        about: workAbout,
+        numberOfPlayers: workNumberOfPlayers,
+        whoseTurn: workWhoseTurn,
+        turnTotals: JSON.parse(JSON.stringify(workTurnTotals)),
+      };
+    });
+  }, []);
+
+  // pressed 1-Player button
+  const pressPlayers1 = useCallback(() => {
+    setGameState(prevGameState => {
+      let workNumberOfPlayers = 1;
+      let workMessage = 'Roll again';
+      let workOptMessage = 'Player 1';
+      return {
```

Comment:
You can probably simplify this method a lot by just doing a `... prevGameState` to get all the `prevGameState` fields, and removing all the work variables and just assigning the fields directly.

Reply:
DONE